library/psa_crypto.c

```diff
@@ -968,9 +1025,20 @@ static  psa_status_t psa_internal_export_key( psa_key_slot_t *slot,
 #endif
             }
             if( export_public_key || PSA_KEY_TYPE_IS_PUBLIC_KEY( slot->type ) )
-                ret = mbedtls_pk_write_pubkey_der( &pk, data, data_size );
+            {
+                if( PSA_KEY_TYPE_IS_RSA( slot->type ) )
+                {
+                    ret = pk_write_pubkey_rsa( &pk, data, data_size );
+                }
+                else
```

Comment:
The two branches of this condition call identical functions. We may eventually have different code paths for RSA and ECC to decrease our reliance on pk, but it's too early for that. Just call `mbedtls_pk_write_pubkey` here.

Reply:
I was thinking to possibly do that in this PR, but since we can agree it's out of scope, I'll merge the implementations.